src/modules/commander/Commander.cpp

```diff
@@ -4058,6 +4066,156 @@ void Commander::battery_status_check()
 	}
 }
 
+void Commander::airspeed_use_check()
+{
+	// assume airspeed sensor is good before starting FW flight
+	bool valid_flight_condition = (status.arming_state == vehicle_status_s::ARMING_STATE_ARMED) &&
+			!status.is_rotary_wing &&
+			!land_detector.landed;
+	bool fault_declared = false;
+	bool fault_cleared = false;
+	if (!valid_flight_condition) {
+		_tas_use_inhibit = false;
+		_time_tas_good_declared = hrt_absolute_time();
+		_time_tas_bad_declared = 0;
+		status.aspd_check_failing = false;
+		status.aspd_fault_declared = false;
+		status.aspd_use_inhibit = false;
+		status.aspd_fail_rtl = false;
+		_time_last_airspeed = hrt_absolute_time();
+		_airspeed_fault_type = new(char[7]);
+		_load_factor_ratio = 0.5f;
+	} else {
+		// The vehicle is flying so use the status of the airspeed innovation check '_tas_check_fail' in
+		// addition to a sanity check using airspeed and load factor and a missing sensor data check.
+
+		// Check if sensor data is missing - assume a minimum 5Hz data rate.
+		bool data_missing = false;
+		if ((hrt_absolute_time() -_time_last_airspeed) > 200_ms) {
+			data_missing = true;
+		}
+
+		// Declare data stopped if not received for longer than 1 second
+		bool data_stopped = false;
+		if ((hrt_absolute_time() -_time_last_airspeed) > 1_s) {
+			data_stopped = true;
+		}
+		_time_last_airspeed = hrt_absolute_time();
+
+		// Check if the airpeed reading is lower than physically possible given the load factor
+		_airspeed_sub.update();
+		_airspeed = _airspeed_sub.get();
+		_sensor_bias_sub.update();
+		_sensor_bias = _sensor_bias_sub.get();
+		float max_lift_ratio = fmaxf(_airspeed.indicated_airspeed_m_s, 0.7f) / fmaxf(_airspeed_stall.get(), 1.0f);
+		max_lift_ratio *= max_lift_ratio;
+		_load_factor_ratio = 0.95f * _load_factor_ratio + 0.05f * (fabsf(_sensor_bias.accel_z) / 9.80665f) / max_lift_ratio;
```

Comment:
@priseborough I may be confused, but won't this algorithm quickly drive to 0?  Or are we expecting _sensor_bias.accel_z to be close to gravity?  I guess my question is weather the _sensor_bias actually includes gravity measurement in it or not.

Reply:
the sensor_bias topic accel_z message is the bias correct accelerometer measurement, so it includes the specific force. it will be around -1g in level flight